### PR TITLE
Tickets/dm35042: Mypy and Pyproject updates

### PR DESCRIPTION
**Pull request by rbovill**
* Type annotations.
* Formatting updates for mypy.
* Had to ignore some checks.

### CODE REVIEW

**Review by mareuter — APPROVED** [review]
Looks fine to me. Have you checked that the scripts in `bin` still get packaged properly after the change?

File: conda/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
 
 build:
-  script: python -m pip install --no-deps --ignore-installed .
+  noarch: generic
+  script: python -m pip install --ignore-installed .
+  entry_points:
+    - auxtel_disabled_enabled = lsst.ts.IntegrationTests.auxtel_disabled_enabled:run_auxtel_disabled_enabled
+    - auxtel_offline_standby = lsst.ts.IntegrationTests.auxtel_offline_standby:run_auxtel_offline_standby
+    - auxtel_prepare_for_flatfield = lsst.ts.IntegrationTests.auxtel_prepare_for_flatfield:run_auxtel_prepare_for_flatfield
+    - auxtel_prepare_for_onsky = lsst.ts.IntegrationTests.auxtel_prepare_for_onsky.:un_auxtel_prepare_for_onsky
```
Comment (mareuter) [inline conda/meta.yaml:16]: This doesn't look right: ...`onsky.:un_auxtel`...

**Review by mareuter — APPROVED** [review]
One typo that needs to be addressed, but looks fine otherwise.

File: conda/meta.yaml (hunk 1 above)
Comment (rbovill) [reply conda/meta.yaml]: Fixed.

**Pull request merged**